mypy/types.py

```diff
@@ -858,17 +858,16 @@ def make_simplified_union(items: List[Type], line: int = -1, column: int = -1) -
                     all_items.append(typ)
             items = all_items
 
-        if any(isinstance(typ, AnyType) for typ in items):
-            return AnyType()
-
         from mypy.subtypes import is_subtype
         removed = set()  # type: Set[int]
         for i, ti in enumerate(items):
             if i in removed: continue
             # Keep track of the truishness info for deleted subtypes which can be relevant
             cbt = cbf = False
             for j, tj in enumerate(items):
-                if i != j and is_subtype(tj, ti):
+                tj_is_anylike = (isinstance(tj, AnyType) or
+                                 (isinstance(tj, Instance) and tj.type.fallback_to_any))
+                if i != j and is_subtype(tj, ti) and not tj_is_anylike:
```

Comment:
Actually, I've changed my mind.  I don't feel strongly about this, but I think the conditional is moderately more readable if `tj_is_anylike` comes after the subtype check.  `tj_is_anylike` isn't true often enough for the perf difference to matter.

If you still prefer it otherwise, I'll switch it.